js/piwik.js

```diff
@@ -592,6 +592,43 @@ if (typeof window.Matomo !== 'object') {
 
             return matches ? matches[1] : url;
         }
+        function isPositiveNumberString(str) {
+            // !isNaN(str) could be used but does not cover '03' (octal) and '0xA' (hex)
+            // nor negative numbers
+            return (/^[1-9][0-9]*(\.[0-9]+)?$/).test(str);
+        }
+        function onlyNumbers(data) {
+            var result = {}, k;
+            for (k in data) {
+                if (data.hasOwnProperty(k)) {
+                    if (isNumber(data[k]) || isPositiveNumberString(data[k])) {
```

Comment:
just fyi likely you would only need to check for `isPositiveNumberString(data[k])` as it would also work for numbers. It's otherwise you apply different standards where `0` is allowed as a number but not as a string `'0'` for example.

Reply:
also you would probably need to define that `onlyPositiveNumbers` or something as it's bit misleading. 
Might be easier to simply call the method `onlyIntegers` and then convert the value directly to an integer here and remove the method `roundNumbers`. 

Simply because next time we need to use `roundNumbers` who knows what values we would accept. Maybe it would accept 0 or negative value and then things start to get complicated. It's almost too "abstract" now where it was doing too many things before :)

Reply:
I would usually say it's not needed to change it because the method is used rarely etc. But there's some danger it will be later reused by others and then it becomes tricky that we accept different kind of values. This we should fix and only consistently accept the same kind of values.

Reply:
I believe this is all covered with below commits now. Thanks for the pointers. Will also be useful for the future. :-)